core/arch/arm/mm/tee_pager.c

```diff
@@ -770,4 +809,38 @@ void tee_pager_add_pages(vaddr_t vaddr, size_t npages, bool unmap)
 		core_tlb_maintenance(TLBINV_UNIFIEDTLB, 0);
 	}
 }
+
+void tee_pager_release_zi(vaddr_t vaddr, size_t size)
+{
+	bool unmaped = false;
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+
+	if ((vaddr & SMALL_PAGE_MASK) || (size & SMALL_PAGE_MASK))
+		panic();
+
+	for (; size; vaddr += SMALL_PAGE_SIZE, size -= SMALL_PAGE_SIZE)
+		unmaped |= tee_pager_release_one_zi(vaddr);
+
+	/* Invalidate secure TLB */
+	if (unmaped)
+		core_tlb_maintenance(TLBINV_UNIFIEDTLB, 0);
+
+	thread_set_exceptions(exceptions);
+}
+
+void *tee_pager_request_zi(size_t size)
```

Comment:
Agree, but this function does a kind of allocation that the pager need to deliver later. If we're over-committing the pager need a way of dealing with that or we'd need to avoid over-committing.

It doesn't have to be fixed now but we need to take care about it at some point.